cmd/system-probe/config/adjust.go

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// adjustConfig makes changes to the raw config based on deprecations and inferences.
+func adjustConfig(cfg config.Config) {
+	deprecateString(cfg, spNS("log_level"), "log_level")
+	deprecateString(cfg, spNS("log_file"), "log_file")
+
+	usmEnabled := cfg.GetBool(smNS("enabled"))
+	dsmEnabled := cfg.GetBool(dsmNS("enabled"))
+	// this check must come first, so we can accurately tell if system_probe was explicitly enabled
+	if cfg.GetBool(spNS("enabled")) &&
+		!cfg.IsSet(netNS("enabled")) &&
+		!usmEnabled &&
+		!dsmEnabled {
+		// This case exists to preserve backwards compatibility. If system_probe_config.enabled is explicitly set to true, and there is no network_config block,
+		// enable the connections/network check.
+		log.Info(deprecationMessage(spNS("enabled"), netNS("enabled")))
+		// ensure others can key off of this single config value for NPM status
+		cfg.Set(netNS("enabled"), true)
+	}
+
+	validateString(cfg, spNS("sysprobe_socket"), defaultSystemProbeAddress, ValidateSocketAddress)
+
+	adjustNetwork(cfg)
+	adjustUSM(cfg)
+	adjustSecurity(cfg)
+}
+
+func validateString(cfg config.Config, key string, defaultVal string, fn func(string) error) {
+	if cfg.IsSet(key) {
+		if err := fn(cfg.GetString(key)); err != nil {
+			log.Errorf("error validating `%s`: %s. using default value of `%s`", key, err, defaultVal)
+			cfg.Set(key, defaultVal)
+		}
+	} else {
+		cfg.Set(key, defaultVal)
+	}
+}
+
+func validateInt(cfg config.Config, key string, defaultVal int, fn func(int) error) {
+	if cfg.IsSet(key) {
+		if err := fn(cfg.GetInt(key)); err != nil {
+			log.Errorf("error validating `%s`: %s. using default value of `%d`", key, err, defaultVal)
+			cfg.Set(key, defaultVal)
+		}
+	} else {
+		cfg.Set(key, defaultVal)
+	}
+}
+
+func validateInt64(cfg config.Config, key string, defaultVal int64, fn func(int64) error) {
+	if cfg.IsSet(key) {
+		if err := fn(cfg.GetInt64(key)); err != nil {
```

Comment:
what's the intention of `fn`?

Reply:
A custom validation function

Reply:
let's document or rename, it is not clear from the function